core/src/main/java/org/apache/accumulo/core/client/impl/Namespaces.java

```diff
@@ -100,50 +106,150 @@ private static ZooCache getZooCache(Instance instance) {
     return namespaceMap;
   }
 
-  public static boolean exists(Instance instance, String namespaceId) {
+  public static boolean exists(Instance instance, Namespace.ID namespaceId) {
     ZooCache zc = getZooCache(instance);
     List<String> namespaceIds = zc.getChildren(ZooUtil.getRoot(instance) + Constants.ZNAMESPACES);
-    return namespaceIds.contains(namespaceId);
-  }
-
-  public static String getNamespaceId(Instance instance, String namespace) throws NamespaceNotFoundException {
-    String id = getNameToIdMap(instance).get(namespace);
-    if (id == null)
-      throw new NamespaceNotFoundException(null, namespace, "getNamespaceId() failed to find namespace");
-    return id;
+    return namespaceIds.contains(namespaceId.canonicalID());
   }
 
+  /**
+   * @deprecated Do not use - String for namespace ID is not type safe. Use {@link #getNamespaceName(Instance, Namespace.ID)}
+   */
+  @Deprecated
```

Comment:
Once in `getNameToIdMap` and once when streaming, yeah, but I don't think it's that big of an issue to transform the internal form to the API/presentation form. The set of tables is relatively small.

An option which wouldn't require it to iterate twice would be to make `getNameToIdMap` a generic method, which accepts a `Function` so it can transform to the value to the requested form:

```java
return Tables.getNameToIdMap(e -> e.canonicalId());
```

Tables.java:
```java
public <T> SortedMap<String, T> getNameToIdMap(Function<Table.ID, ? extends T> valueConverter) {
    // ...
}
```